### ISSUE TITLE
JPG support for screenshots

### ISSUE THREAD

**Issue by seriousgigi** [xxxily/h5player#596]
hello, this script used to save screenshots as JPG instead PNG before v3.4.4

can we please get JPG/PNG option with new version?

i take a lot of screenshots and this is the reason why i started to use this script in the beginning but size of PNG screenshots are 4x bigger than JPG ones. 

please make JPG default again or give us an option to do that.

**Comment by xxxily** [comment]
目前默认就是JPG 格式，具体代码在这：https://github.com/xxxily/h5player/blob/master/dist/h5player.js#L3347

**Comment by seriousgigi** [comment]
> 目前默认就是JPG 格式，具体代码在这：https://github.com/xxxily/h5player/blob/master/dist/h5player.js#L3347

it still downloads in PNG format.

1080p video with jpg size is around 500-600 KB while PNG is around 2.5 MB

**Comment by xxxily** [comment]
> > 目前默认就是JPG 格式，具体代码在这：https://github.com/xxxily/h5player/blob/master/dist/h5player.js#L3347
> 
> it still downloads in PNG format.
> 
> size of jpg screenshot in 1080p video is around 500-600 KB while PNG is around 2.5 MB
> 
> please take a screenshot with v3.4.4 and current one. you will see the bug.
> 
> apart from this issue there is another one;
> 
> sometimes this script doesnt download screenshots instead it opens screenshots on a new tab. and that new tab can only be saved in PNG format. please look into this too.

脚本的最新版本是v3.7.11，请使用最新版本进行截图，如果还有问题，请提供具体的网址，以便我这边进行复现。

**Comment by seriousgigi** [comment]
i am already using v3.7.11 and it saves as PNG.

do you want me to upload a video to proof?

**Comment by xxxily** [comment]
> i am already using v3.7.11 and it saves as PNG. i already told you v3.4.4 is the last one works. from that version all of new versions are saving as PNG
> 
> do you want me to upload a video to proof?

是的，请提供更多有效的信息，以便我这边进行复现，例如浏览器版本，具体的网站地址，操作步骤等，没复现的话，没法修复该bug

**Comment by seriousgigi** [comment]
no need for specific websites since it doesnt work at any video streaming sites such as youtube or chaturbate for example.
i use opera browser, windows 10 and using your script with tampermonkey's latest version.

using "shift + s" to capture screenshots. 

if these are no enough i can upload a video later.

**Comment by xxxily** [comment]
> no need for specific websites since it doesnt work at any video streaming sites such as youtube or chaturbate for example. i use opera browser, windows 10 and using your script with tampermonkey's latest version.
> 
> using "shift + s" to capture screenshots.
> 
> if these are not enough i can upload a video later.

建议你换edge或chrome浏览器试试，目前最新的脚本的代码中默认就是JPG格式的，在youtube 上使用"shift + s" 进行截图默认存的就是JPG格式的截图，如果换了浏览器还是不正常，你上传下操作的视频看下是什么问题吧，我这边实在没法复现你这个问题

**Comment by seriousgigi** [comment]
let me say smth i just realized. the type is JPG (or JPEG) as you said but the file size is 4x bigger than usual jpg because it is not compressed. 1920x screenshot cant be larger than 1mb. it's size is usually around 500-600 KB

![Screenshot (441) (2jpg)](https://github.com/xxxily/h5player/assets/83796753/9eac2328-0665-4d10-afdd-b105d2aca6d3)

you can see the file is actually PNG by opening it on pro image viewer like acdsee. when i open this jpg screenshot on acdsee it says it is png. so the jpg is just on the name not reality.

![Screenshot (442) (2jpg)](https://github.com/xxxily/h5player/assets/83796753/a26984a9-a4b6-492b-a951-0340346a5f5f)

if you dont believe just do it by yourself.

1. take a screenshot of any video on youtube with latest version.
2. downgrade you script to v3.4.4 and then take same screenshot with this version
3. compare file sizes of both files. you will see the screenshot taken with v3.4.4 is around 600 KB while other is around 2 MB

**Comment by xxxily** [comment]
> let me say smth i just realized. the type is JPG (or JPEG) as you said but the file size is 3-4x bigger than usual jpg because it is not compressed. 1920x screenshot cant be larger than 1mb. it's size is usually around 500-600 KB
> 
> ![Screenshot (441) (2jpg)](https://private-user-images.githubusercontent.com/83796753/297097868-9eac2328-0665-4d10-afdd-b105d2aca6d3.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU0NTQxNDksIm5iZiI6MTcwNTQ1Mzg0OSwicGF0aCI6Ii84Mzc5Njc1My8yOTcwOTc4NjgtOWVhYzIzMjgtMDY2NS00ZDEwLWFmZGQtYjEwNWQyYWNhNmQzLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE3VDAxMTA0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVlNDZlNWUyNTVhNGQ3YWE2ZGRmYmVhMDI3NGViMTQ5ZDlkYmQ4MjEzNjNiODFhN2I1YmI4NDE2ZWZlZDk2NjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5Dfy5eHO1rlXxfBvcGeVM8df_q4hm_FMe7ouxborEYg)
> 
> you can see the file is actually PNG by opening it on pro image viewer like acdsee. when i open this jpg screenshot on acdsee it says it is png. so the jpg is just on the name not reality.
> 
> ![Screenshot (442) (2jpg)](https://private-user-images.githubusercontent.com/83796753/297097881-a26984a9-a4b6-492b-a951-0340346a5f5f.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU0NTQxNDksIm5iZiI6MTcwNTQ1Mzg0OSwicGF0aCI6Ii84Mzc5Njc1My8yOTcwOTc4ODEtYTI2OTg0YTktYTRiNi00OTJiLWE5NTEtMDM0MDM0NmE1ZjVmLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE3VDAxMTA0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxZWU2Y2FjOTQ4NTVhMjFjNTZkN2VlMmViZWM5Y2NhMTJlMWYyNWZlMTUzYjk4N2Q5OGE0NzZkMWQ5YWE2NGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xPvLpf9TffSvuzQEv8_ls5VeS2oksI5BoOzZoCPaBp4)
> 
> note: if i convert your screenshot with 90 quality setting to jpg the file size is reducing to 350 KB.
> 
> if you dont believe just test it by yourself;
> 
> 1. take a screenshot of any video on youtube with latest version.
> 2. downgrade your script to v3.4.4 and then take same screenshot with this version
> 3. compare file sizes of both files. you will see the screenshot taken with v3.4.4 is around 600 KB while other is around 2 MB

感谢您的耐心反馈，已经理解你的意思，也排查出了bug的原因，下个版本将修复该问题

**Comment by seriousgigi** [comment]
hello! thx for latest update that fixed the problem.

only one problem left. i dont know how to explain but sometimes on some sites or some video players, when you take screenshot it opens new tab and shows screenshot in that new tab instead saving the screenshot. and when it does that the image is always in PNG format.

i can give an example for this but the site i am trying is p*rn site. so i dont know if sharing that link is allowed here...

**Comment by seriousgigi** [comment]
yes but it again saves screenshots in PNG format in this way. can you change that to jpg like you did for last time? it would be amazing if you pull this off :)

**Comment by seriousgigi** [comment]
also in that way screenshots are saved with name: "download" instead video title and time.
is there anything you can do to change this behavior?

**Comment by seriousgigi** [comment]
hello in recent version it started to save screenshots as PNG again. can you tell me how to fix this again? thx.
